### ISSUE TITLE
Entangle syncing wrong data.

### ISSUE THREAD

**Issue by cihantas** [livewire/livewire#1781]
### Description

Using @entangle in a nested Alpine component with Livewire causes sync problems. We have a simple checkbox with `x-data="{ enabled: @entangle($attributes->wire('model')) }>`.
Clicking the checkbox causes `enabled` to switch from a `boolean` to an JS `Object`.

I tried looking into the Object using `<h1 x-text="JSON.stringify(enabled)"></h1>`, but it's causing a circular lookup error. If I evaluate the entangle Javascript code `window.Livewire.find('rtaric01yz9EpxpnzZcI').entangle('enabled')` I can see that defer is indeed pointing to the parent object again.

![image](https://user-images.githubusercontent.com/1579976/95453423-1f86a300-096b-11eb-9bbe-dedeb77b801c.png)

However, I was able to print the keys with `<h1 x-text="JSON.stringify(Object.keys(enabled))"></h1>`. The result is `["isDeferred","livewireEntangle","defer"]`. As we can also see in the screenshot above.

### Stripped-down, copy-pastable code snippets

**Parent Component**
```html
<div x-data="{ open: @entangle('open') }">
      <x-inputs.toggle wire:model="enabled"/>
</div>
```

**Checkbox Component**
```html
<div v-pre x-data="{ enabled: @entangle($attributes->wire('model')) }"
     x-on:click.prevent.stop="enabled = !enabled">

    <h1 x-text="JSON.stringify(Object.keys(enabled))"></h1>
    {{-- <h1 x-text="JSON.stringify(enabled)"></h1> --}}

</div>
```

Please excuse the rather weak information. I'll try to provide more later.

**Context**
- Livewire version: [e.g. 2.2.9]
- Laravel version: [e.g. 8.3.1]
- Browser: [e.g. Chrome, Safari]

**Comment by basepack** [comment]
Hi @cihantas, can you also show us the code of your Livewire component?

**Comment by cihantas** [comment]
Sure, sorry for the delay.

```php
class Stage extends Component
{
    // Filled on mount.    
    public $enabled;
    public $stageId;

    public $open = false;

    public function mount(AcrStage $stage)
    {
        $this->fill($stage);
        $this->stageId = $stage->id;
    }

    public function getStageProperty()
    {
        return AcrStage::find($this->stageId);
    }

    public function updatedEnabled($value): void
    {
        $this->stage->forceFill([
            'enabled' => $value,
        ])->save();
    }
}

```

**Comment by basepack** [comment]
Thanks! Indeed your bug description seems correct. 

Just a heads up, with Livewire 2.0 you can rewrite your setup like this:
```php
class Stage extends Component
{
    public AcrStage $stage; // I assume an Eloquent model ArcStage
    public $open = false;

    public function save(): void
    {
        $this->stage->save();
    }
}
```

You could even remove that `updateEnabled` function if you change it like this:
```blade
<div x-data="{ open: @entangle('open') }">
      <x-inputs.toggle wire:model="stage.enabled"/> <!-- changed-->
      <button wire:click="save">Save</button> <!-- changed-->
</div>
```

Of course you have to add that `save` function to your component, and I assumed ArcStage is an Eloquent model.

And it then should works (yea, when the `@entangle` bugs are fixed).

**Comment by basepack** [comment]
@calebporzio I added a LaravelPlayground where this issue is shown: https://laravelplayground.com/#/snippets/1f39adf0-4918-49c8-a5be-4c7b402878b2

Gif to show you the most important parts of this setup:
![example](https://user-images.githubusercontent.com/939500/95557114-9fb51300-0a14-11eb-9a0d-8890dcadcb16.gif)

**Comment by cihantas** [comment]
@basepack Thanks for the heads up. 👍  We're aware of the cleaner approaches. There's a lot more code involved than visible here, so our migration will take some time, but we're on it. :-)

**Comment by calebporzio** [comment]
Thanks for making this more reproducible @basepack - I tested this and it will be fixed in the next release of Livewire v2.3.0 - thanks!
